tests/test_module_loader.py

```diff
@@ -257,3 +257,46 @@ def test_load_module_v1_and_v2_installed(
     assert f"Module {module_name} is installed as a V1 module and a V2 module: V1 will be ignored." in caplog.text
     assert module_name in project.modules
     assert isinstance(project.modules[module_name], ModuleV2)
+
+
+def test_load_module_recursive_v2_module_depends_on_v1(local_module_package_index: str, snippetcompiler) -> None:
```

Comment:
Could you add a test for the other way around (v1 depends on v2)?

Reply:
This test is already covered by https://github.com/inmanta/inmanta-core/blob/59a5c1a5b250744148eb6e950d249fd46578d851/tests/test_module_loader.py#L115. Do you think that is sufficient?

Reply:
I guess so. There's also the case where the v2 has not been preinstalled but that should be covered by the tests I'm working on.